test/functional/shared.js

```diff
@@ -57,8 +98,26 @@ function setupDatabase(configuration, dbsToClean) {
     );
 }
 
-function makeCleanupFn(client) {
-  return function(err) {
+/**
+ * Safely perform a test with provided MongoClient, ensuring client won't leak.
+ *
+ * @param {MongoClient} [client] if not provided, withClient must be bound to test function `this`
+ * @param {Function} operation (client):Promise or (client, done):void
+ * @param {Function} [errorHandler]
```

Comment:
I don't see `errorHandler` being used anywhere, do we need it? Seems to me that just adding an `expect(client).to.exist` is enough to tell us if something went wrong, I'm not seeing the use case for returning control back to the caller.